### DISCUSSION TITLE
Posture issues with data writing and reading

### DISCUSSION THREAD

**Discussion by xiaozhuang-a** [milvus-io/milvus#25268] [Q&A and General discussion] [answered]
When there are a large number of collections, it is found that the ATTU interface will load for a very long time. Therefore, I would like to ask if there is any negative impact of inserting collections that do not actively flush and release when the number of records in the collection is small and the number of collections is large? Why? The premise is that the querynode resource memory is sufficient for use

**Comment by yhmo** [comment]
Milvus allocate 1024 rows space for each "shard" when you carete a new collection. The pre-allocated memory space depends on dimension and shard number. Assuming you create a collection with shard_num=2, vector dimension is 512. The pre-allocated memory size is 512*4*1024*2 = 4MB. So, if you create 1000 this kind of collections, it will require at least 4GB memory size.

**Reply by xiaozhuang-a** [reply]
Thank you for your answer. I still don't understand.
Where are these 1024 row spaces stored, s3? Memory?

**Reply by yhmo** [reply]
It is in-memory buffer.
It pre-allocates 1024 rows even you have no data inserted. Once the insert data exceed 1024 rows, it re-allocate a larger size, just like the std::vector class in C++. After flush() is called or triggered by certain conditions, the in-memory buffer is flushed into storage(s3/minio), and it allocates a new 1024 rows buffer for your next insert operation.

**Reply by yhmo** [reply, marked as answer]
It is an in-memory buffer.
It pre-allocates 1024 rows even if you have no data inserted. Once the insert data exceed 1024 rows, it re-allocates a larger size, just like the std::vector class in C++. After flush() is called or triggered by certain conditions, the in-memory buffer is flushed into storage(s3/minio), and it allocates a new 1024 rows buffer for your next insert operation.

**Reply by xiaozhuang-a** [reply]
Thank you for your answer.
